core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala

```diff
@@ -859,6 +859,44 @@ class SparkSubmitSuite
     }
   }
 
+  test("Avoid re-upload remote resources in yarn client mode") {
+    val hadoopConf = new Configuration()
+    hadoopConf.set("fs.s3a.impl", "org.apache.spark.deploy.TestFileSystem")
+    hadoopConf.set("fs.s3a.impl.disable.cache", "true")
```

Comment:
This pattern is getting really repetitive, can you put this code in its own method and call that from existing tests?